### PR TITLE
typing: Add type hints to sentry/utils/services.py

### PR DESCRIPTION
**Pull request by wedamija**
We use these service backends in a lot of places that are typed. Adding typing here adds a lot of
value since we can now properly introspect things that inherit from `Service` and related.

### CODE REVIEW

File: src/sentry/processing/realtime_metrics/__init__.py [hunk 1]
```diff
@@ -6,17 +6,17 @@
 
 from .base import RealtimeMetricsStore
 
-realtime_metrics_store: RealtimeMetricsStore = LazyServiceWrapper(
```
Comment (wedamija) [inline src/sentry/processing/realtime_metrics/__init__.py:9]: This wasn't the right type before and caused some confusion for mypy. Moved this type down the the `TYPE_CHECKING` area.

File: tests/sentry/tasks/test_low_priority_symbolication.py [hunk 2]
```diff
@@ -234,11 +235,13 @@ def test_normal_rate_long_duration(self) -> None:
     def test_low_rate_long_duration(self) -> None:
         # 1 event/m for 3m, 9m durations
         histograms = []
+        durations = None
```
Comment (wedamija) [inline tests/sentry/tasks/test_low_priority_symbolication.py:238]: pyright started complaining about unbound variable issues here

File: src/sentry/utils/types.py [hunk 3]
```diff
@@ -167,12 +169,11 @@ def convert(self, value):
...
-except NameError:  # long was removed in Python 3
-    pass
 
 
 def type_from_value(value):
     """Fetch Type based on a primitive value"""
     return _type_mapping[type(value)]
+
+
+AnyCallable = Callable[..., Any]
```
Comment (wedamija) [inline src/sentry/utils/types.py:179]: Not sure if this is really necessary, but it's a bit less to type

File: src/sentry/utils/services.py [hunk 4]
```diff
@@ -135,8 +157,6 @@ def copy(self):
     None,
 ]
 
-T = TypeVar("T")
```
Comment (wedamija) [inline src/sentry/utils/services.py:138]: It seems like we always pass in `Service` backends here so seemed to make more sense to be explicit.

File: src/sentry/processing/realtime_metrics/__init__.py (hunk 1 above)
Comment (flub) [reply src/sentry/processing/realtime_metrics/__init__.py:9]: oh, thanks for figuring this out.  i remember being confused why mypy was still not handling this.

**Review by flub — APPROVED** [review]

**Pull request merged**